### PR TITLE
Replace package `ocramius/package-versions` to internal service

### PR DESCRIPTION
**Pull request by gomzyakov**
### Changed

- Package `ocramius/package-versions` replaced to internal service `Versions`

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I wrote unit tests for my code _(if tests is required for my changes)_
- [x] I have made changes in `CHANGELOG.md` file

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,12 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog][keepachangelog] and this project adheres to [Semantic Versioning][semver].
 
+## Unreleased
+
+### Changed
+
+- Package `ocramius/package-versions` replaced to internal service `Versions`
```
Comment (gomzyakov) [inline CHANGELOG.md:11]: ### Why?

Any version of package `ocramius/package-versions` supports only 2 versions of PHP.

Using our own service, we will be able to support a wider set of PHP versions.

File: src/Services/Versions.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+final class Versions
+{
+    /**
+     * Returns the full version of the specified package as `v1.4.0@5a29c10`.
+     *
+     * @param string $packageName
+     *
+     * @return string
+     */
+    public static function getVersion(string $packageName): string
```
Comment (Zaboday) [inline src/Services/Versions.php]: Need snake case here: `$packageName` -> `$package_name`.

File: src/Specifications.php [hunk 3]
```diff
@@ -6,8 +6,8 @@
 
 use Exception;
 use InvalidArgumentException;
-use PackageVersions\Versions;
 use Illuminate\Support\Collection;
+use Avtocod\Specifications\Services\Versions;
```
Comment (Zaboday) [inline src/Specifications.php]: Can we have 🎄 here?

Like that:
```
use Illuminate\Support\Collection;
use Avtocod\Specifications\Structures\Field;
use Avtocod\Specifications\Services\Versions;
```

File: tests/SpecificationsTest.php [hunk 4]
```diff
@@ -8,9 +8,9 @@
 use ReflectionClass;
 use ReflectionMethod;
 use InvalidArgumentException;
-use PackageVersions\Versions;
 use PHPUnit\Framework\TestCase;
 use Illuminate\Support\Collection;
+use Avtocod\Specifications\Services\Versions;
```
Comment (Zaboday) [inline tests/SpecificationsTest.php]: Can we have 🎄 here?

File: src/Services/Versions.php (hunk 2 above)
Comment (gomzyakov) [reply src/Services/Versions.php]: Fixed

File: src/Specifications.php (hunk 3 above)
Comment (gomzyakov) [reply src/Specifications.php]: Yep, fixed

File: tests/SpecificationsTest.php (hunk 4 above)
Comment (gomzyakov) [reply tests/SpecificationsTest.php]: Yep, fixed

File: src/Services/Versions.php [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+declare(strict_types=1);
+
+namespace Avtocod\Specifications\Services;
+
+use Composer\InstalledVersions;
+
+/**
+ * Wrapper for Composer runtime utility `InstalledVersions`.
+ */
+final class Versions
```
Comment (gomzyakov) [inline src/Services/Versions.php:12]: Where are the tests?

Comment (gomzyakov) [reply src/Services/Versions.php:12]: Tests added https://github.com/avtocod/specs-php/pull/16/commits/7325fe04296007d0d406cff75b0a2b2e31289669

File: tests/Services/VersionsTest.php [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+use function json_decode;
+use function array_merge;
+use function file_get_contents;
+
+/**
+ * @group Services
+ * @group Services_Versions
+ *
+ * @covers \Avtocod\Specifications\Services\Versions<extended>
```
Comment (Zaboday) [inline tests/Services/VersionsTest.php]: Please, replace by `@covers \Avtocod\Specifications\Services\Versions`

File: tests/Services/VersionsTest.php [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+use Illuminate\Support\Str;
+use PHPUnit\Framework\TestCase;
+use Avtocod\Specifications\Services\Versions;
+
+use function json_decode;
+use function array_merge;
+use function file_get_contents;
+
+/**
+ * @group Services
```
Comment (Zaboday) [inline tests/Services/VersionsTest.php]: Lets stay with one group: "@group versions"

File: tests/Services/VersionsTest.php [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    /**
+     * Get all packages data from `composer.lock` file.
+     *
+     * @return array
+     */
+    private function getAllPackagesData(): array
+    {
+        $lock_data = json_decode(file_get_contents(__DIR__ . '/../../composer.lock'), true);
```
Comment (Zaboday) [inline tests/Services/VersionsTest.php]: Add `JSON_THROW_ON_ERROR hear:
```php
$lock_data = json_decode(file_get_contents(__DIR__ . '/../../composer.lock'), true, 512, \JSON_THROW_ON_ERROR);
```

Comment (gomzyakov) [reply tests/Services/VersionsTest.php]: Thanks for the note! `JSON_THROW_ON_ERROR` added.

File: tests/Services/VersionsTest.php (hunk 7 above)
Comment (gomzyakov) [reply tests/Services/VersionsTest.php]: Ok, accepted!

File: tests/Services/VersionsTest.php (hunk 6 above)
Comment (gomzyakov) [reply tests/Services/VersionsTest.php]: Replaced

**Pull request merged**